src/array.jl

```diff
@@ -972,3 +963,29 @@ Base.repeat(a::CatArrOrSub{T, N},
 Base.repeat(a::CatArrOrSub{T, N};
             inner = nothing, outer = nothing) where {T, N} =
     CategoricalArray{T, N}(repeat(refs(a), inner=inner, outer=outer), copy(pool(a)))
+
+# JSON3 writing/reading
+StructTypes.StructType(::Type{<:CategoricalVector}) = StructTypes.ArrayType()
+
+StructTypes.construct(::Type{<:CategoricalArray}, array::Vector) =
+    categoricalgeneral(array)
+function categoricalgeneral(array)
```

Comment:
```suggestion
function StructTypes.construct(::Type{<:CategoricalArray}, array::Vector)
```
Also the convention used in the package is to use `A` for array arguments rather than `array`.